core/interfaces/i_procedure_block.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a procedure block.
+ */
+
+'use strict';
+
+goog.provide('Blockly.Procedures.ProcedureBlock');
```

Comment:
Make this interface `Blockly.IProcedureBlock` instead--that way it's not adding something onto a namespace that's defined somewhere else.

If we need to keep the `Blockly.Procedures.ProcedureBlock` typedef around you can do a typedef in procedures.js, where it was previously, to make it equal to `IProcedureBlock`.

An interface needs to be required if there is code that is implementing it (`@implements`), but otherwise can just be used with `requireType`.